src/pages/data-source/[id].vue

```diff
@@ -313,22 +318,35 @@ function getPrev() {
   return searchStore.mostRecentSearchIds[previousIdIndex.value];
 }
 
-onMounted(() => {
-  handleShowMoreButton();
-  window.addEventListener('resize', handleShowMoreButton);
-});
+let observer = null;
 
-onUnmounted(() => {
-  window.removeEventListener('resize', handleShowMoreButton);
-});
+watchEffect(() => {
+  if (!descriptionRef.value) return;
+
+  const el = descriptionRef.value;
+
+  const checkOverflow = () => {
+    showExpandDescriptionButton.value = el.offsetHeight < el.scrollHeight;
+  };
 
-function handleShowMoreButton() {
-  if (descriptionRef.value?.offsetHeight < descriptionRef.value?.scrollHeight) {
-    showExpandDescriptionButton.value = true;
-  } else {
-    showExpandDescriptionButton.value = false;
+  observer = new ResizeObserver(checkOverflow);
+  observer.observe(el);
+
+  // Also run on initial mount
+  checkOverflow();
+});
+watch(
+  () => route.params.id,
+  () => {
+    isDescriptionExpanded.value = false;
   }
-}
+);
+onBeforeUnmount(() => {
+  if (observer && descriptionRef.value) {
+    observer.unobserve(descriptionRef.value);
+    observer.disconnect();
```

Comment:
then resetting to `null` in the cleanup

```suggestion
    observer.unobserve(descriptionRef.value);
    observer.disconnect();
    observer.value = null;
```